src/simplewallet/simplewallet.cpp

```diff
@@ -2260,9 +2260,16 @@ bool simple_wallet::set_default_ring_size(const std::vector<std::string> &args/*
     }
  
     if (ring_size != 0 && ring_size != DEFAULT_MIX+1)
-      message_writer() << tr("WARNING: this is a non default ring size, which may harm your privacy. Default is recommended.");
-    else if (ring_size == DEFAULT_MIX)
-      message_writer() << tr("WARNING: from v8, ring size will be fixed and this setting will be ignored.");
+    {
+      if (m_wallet->use_fork_rules(8, 0))
+      {
+        message_writer() << tr("WARNING: from v8, ring size will be fixed and this setting will be ignored.");
+      }
+      else
+      {
+        message_writer() << tr("WARNING: this is a non default ring size, which may harm your privacy. Default is recommended.");
+      }
+    }
```

Comment:
Can you explain why this part ? It seems a bit counter productive )ie, you've got to have a daemon to ask for fork info, and we'll never be at less than 8 anyway.

Reply:
The original code
```c++
    if (ring_size != 0 && ring_size != DEFAULT_MIX+1)
      message_writer() << tr("WARNING: this is a non default ring size, which may harm your privacy. Default is recommended.");
    else if (ring_size == DEFAULT_MIX)
      message_writer() << tr("WARNING: from v8, ring size will be fixed and this setting will be ignored.");
```
was made in https://github.com/monero-project/monero/commit/5ffb2ff9b7c301eda5811a939c705f26627c4735 where fixed ring size was introduced, but the network was still on v7. I overlooked this point when I was reviewing the PR #4219, but now I find the original code doesn't make sense, because the `else if (ring_size == DEFAULT_MIX)` branch will never be executed (`ring_size == DEFAULT_MIX` is already caught by `ring_size != DEFAULT_MIX+1`). So my proposed change here is intended to fix that logic so that a proper warning message will be shown depending on the consensus version.

But you're correct, this branching is no longer needed for Monero because it's already on v8. I made this patch along with the corresponding change in Aeon https://github.com/aeonix/aeon/pull/136/commits/78a1d4519b579eadf312839fa76e38bc05252ae9#diff-cb82e0aa6db499004305bf35f0eaafbc which is going to switch to fixed ring size in the upcoming fork. If you don't like this branch, I can remove it of course.

Reply:
I see. The second branch should have been using DEFAULT_MIX+1.
The fork check might be broken without a daemon (ie, cold signing). If it still works, I'm OK either way. If it breaks, it needs removing.

Reply:
> The second branch should have been using DEFAULT_MIX+1.

`ring_size==DEFAULT_MIX+1` means the appropriate value is specified; why would you want to display the warning message?

> The fork check might be broken without a daemon (ie, cold signing).

I don’t think your concern applies here, since this setting is relevant only for daemon-connected watch-only wallets initiating txes; offline full wallets don’t need to worry about what ring size to use.